File: java/src/main/java/problems1_100/LengthOfLastWord.java
package problems1_100;

enum LengthOfLastWord {
    DEFAULT {
        @Override
        int lengthOfLastWord(String s) {
            char[] chars = s.toCharArray();
            for (int i = 0; i < chars.length / 2; i++) {
                char temp = chars[i];
                chars[i] = chars[chars.length - 1 - i];
                chars[chars.length - 1 - i] = temp;
            }

            int count = 0;
            boolean startCounting = false;
            for (char c : chars) {
                if (c == ' ') {
                    if (startCounting) {
                        break;
                    }
                } else {
                    startCounting = true;
                }
                if (startCounting) {
                    count++;
                }
            }
            return count;
        }
    };

    abstract int lengthOfLastWord(String s);
}

File: kotlin/src/main/kotlin/problems1_100/LengthOfLastWord.kt
package problems1_100

enum class LengthOfLastWord {
    DEFAULT {
        override fun lengthOfLastWord(s: String): Int {
            val chars = s.toCharArray()
            for (i in 0 until chars.size / 2) {
                val temp = chars[i]
                chars[i] = chars[chars.size - 1 - i]
                chars[chars.size - 1 - i] = temp
            }

            var count = 0
            var startCounting = false
            for (c in chars) {
                if (c == ' ') {
                    if (startCounting) {
                        break
                    }
                } else {
                    startCounting = true
                }
                if (startCounting) {
                    count++
                }
            }
            return count
        }
    },
    ;

    abstract fun lengthOfLastWord(s: String): Int
}
